src/raftstore/store/peer.rs

```diff
@@ -775,6 +775,18 @@ impl Peer {
     }
 }
 
+fn is_tranfer_leader_allowed<T: Storage>(raw_node: &RawNode<T>, peer_id: u64) -> bool {
+    let status = raw_node.status();
+    for progress in status.progress.values() {
+        if progress.state == ProgressState::Snapshot {
+            return false;
+        }
+    }
+
+    let last_index = raw_node.raft.raft_log.last_index();
```

Comment:
I don't think it is a good way to use too deep level raft and raft_log in store here. 
Please see how gc gets last_index in on_raft_gc_log_tick.